### PR TITLE
Bindings for Text2Image pipeline.

### PR DESCRIPTION
**Pull request by popovaan**
Bindings and Python sample for Text2Image pipeline.
Ticket: CVS-152323

### CODE REVIEW

File: src/python/py_lora_adapter.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+namespace py = pybind11;
+namespace utils = ov::genai::pybind::utils;
+
+
+void init_lora_adapter(py::module_& m) {
+    py::class_<ov::genai::Adapter>(m, "Adapter", "Inmutable LoRA Adapter that carries the adaptation matrices and serves as unique adapter identifier.")
+        .def(py::init([](
+            const std::string& path
+        ) {
+            return std::make_unique<ov::genai::Adapter>(path);
```
Comment (slyalin) [inline src/python/py_lora_adapter.cpp:21]: The `Adapter` class already behaves as a smart pointer like `std::shared_ptr`. Do we need an extra level of indirection with `unique_ptr`?

Comment (popovaan) [reply src/python/py_lora_adapter.cpp]: Agree, not needed. Removed `unique_ptr`.

File: samples/python/text2image/text2image.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    device = 'CPU'  # GPU can be used as well
+    random_generator = Generator(50)  # openvino_genai.CppStdGenerator can be used
+    pipe = openvino_genai.Text2ImagePipeline(args.model_dir, device)
+    image_tensor = pipe.generate(
+        args.prompt,
+        width=512,
+        height=512,
+        num_inference_steps=20,
+        num_images_per_prompt=1,
```
Comment (slyalin) [inline samples/python/text2image/text2image.py:35]: If I don't specify this, what happens?

File: src/python/py_lora_adapter.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            return std::make_unique<ov::genai::AdapterConfig>(adapters, mode);
+        }),
+        py::arg("adapters"),
+        py::arg("mode") = ov::genai::AdapterConfig::Mode::MODE_AUTO);
+
+    //  TODO: Need bindings for following methods
+
+    // template <typename AT, typename std::enable_if<std::is_constructible<Adapter, AT>::value, bool>::type = true>
+    // AdapterConfig (const std::initializer_list<AT>& adapters, Mode mode = MODE_AUTO) : AdapterConfig(std::vector<Adapter>(adapters), mode) {}
+    // AdapterConfig (const std::initializer_list<std::pair<Adapter, float>>& adapters, Mode mode = MODE_AUTO) : AdapterConfig(std::vector<std::pair<Adapter, float>>(adapters), mode) {}
```
Comment (slyalin) [inline src/python/py_lora_adapter.cpp:82]: We don't need these variants in Python. In C++ they were required to enable a little bit more syntax sugar and avoid ambiguous overloading. In Python we will pass it as natural list and tuple combination and it works out-of-the-box.

File: src/python/py_lora_adapter.cpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    // operator bool() const
```
Comment (slyalin) [inline src/python/py_lora_adapter.cpp:83]: Looks like it should be automatically defined.

File: samples/python/text2image/text2image.py (hunk 2 above)
Comment (popovaan) [reply samples/python/text2image/text2image.py:35]: Generated this image.
lora

File: src/python/py_lora_adapter.cpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    py::class_<ov::genai::Adapter>(m, "Adapter", "Inmutable LoRA Adapter that carries the adaptation matrices and serves as unique adapter identifier.")
+        .def(py::init([](
+            const std::string& path
+        ) {
+            return ov::genai::Adapter(path);
+        }),
+        py::arg("path"), "path", 
+        R"(
+            Inmutable LoRA Adapter that carries the adaptation matrices and serves as unique adapter identifier.
+            path (str): Path.
```
Comment (slyalin) [inline src/python/py_lora_adapter.cpp:26]:
```suggestion
            path (str): Path to adapter file in safetensors format.
```

File: src/python/py_lora_adapter.cpp [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        )"); 
+        // TODO: operator bool() const 
+
+    auto adapter_config = py::class_<ov::genai::AdapterConfig>(m, "AdapterConfig", "Adapter config.");
```
Comment (slyalin) [inline src/python/py_lora_adapter.cpp:30]:
```suggestion
    auto adapter_config = py::class_<ov::genai::AdapterConfig>(m, "AdapterConfig", "Adapter config that defines a combination of LoRA adapters with blending parameters.");
```

File: src/python/py_lora_adapter.cpp (hunk 4 above)
Comment (popovaan) [reply src/python/py_lora_adapter.cpp:83]: Added `__bool__` operator as agreed, works fine.

**Comment by ilya-lavrenov** [comment]
Since, you have added LoRA support in this PR, it would great also to add LoRA sample based on LLMPipeline similar to https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/cpp/lora_greedy_causal_lm/lora_greedy_causal_lm.cpp

But, I think we need to drop explicit folder `lora_greedy_causal_lm` and re-use `greedy_causal_lm` as we have done for text to image (keep two sample files within the same folder).

It can be done in a separate PR 🍇

File: src/python/py_text2image_pipeline.cpp [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+    :param prompt: input prompt
+    :type prompt: str
+
+    :param kwargs: arbitrary keyword arguments with keys corresponding to generate params. 
+    
+    Expected parameters list: 
+    prompt_2: str - second prompt,
+    prompt_3: str - third prompt,
+    negative_prompt: str - negative prompt,
+    negative_prompt_2: str - second negative prompt,
```
[the remark this answers is not in the document]
Comment (popovaan) [reply src/python/py_text2image_pipeline.cpp:43]: Done.

File: src/python/py_text2image_pipeline.cpp [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+        py::object value = py::cast<py::object>(item.second);
+
+        if (key == "prompt_2") {
+            config.prompt_2 = py::cast<std::string>(value);
+        } else if (key == "prompt_3") {
+            config.prompt_3 = py::cast<std::string>(value);
+        } else if (key == "negative_prompt") {
+            config.negative_prompt = py::cast<std::string>(value);
+        } else if (key == "negative_prompt_2") {
+            config.negative_prompt_2 = py::cast<std::string>(value);
```
[the remark this answers is not in the document]
Comment (popovaan) [reply src/python/py_text2image_pipeline.cpp:72]: Done.

File: src/python/py_text2image_pipeline.cpp [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+        std::string key = py::cast<std::string>(item.first);
+        py::object value = py::cast<py::object>(item.second);
+
+        if (key == "prompt_2") {
+            params.insert({ov::genai::prompt_2(std::move(py::cast<std::string>(value)))});
+        } else if (key == "prompt_3") {
+            params.insert({ov::genai::prompt_3(std::move(py::cast<std::string>(value)))});
+        } else if (key == "negative_prompt") {
+            params.insert({ov::genai::negative_prompt(std::move(py::cast<std::string>(value)))});
+        } else if (key == "negative_prompt_2") {
```
[the remark this answers is not in the document]
Comment (popovaan) [reply src/python/py_text2image_pipeline.cpp:110]: Done.

File: samples/python/text2image/.gitattributes [hunk 10]
```diff
@@ -0,0 +1,3 @@
+baseline.bmp filter=lfs diff=lfs merge=lfs -text
+image.bmp filter=lfs diff=lfs merge=lfs -text
+lora.bmp filter=lfs diff=lfs merge=lfs -text
```
[the remark this answers is not in the document]
Comment (popovaan) [reply samples/python/text2image/.gitattributes]: Looks not needed. Removed.

File: samples/python/text2image/lora.py [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+                          random_generator=Generator(42),
+                          width=512,
+                          height=896,
+                          num_inference_steps=20)
+
+    image_write("lora.bmp", image)
+    print("Generating image without LoRA adapters applied, resulting image will be in baseline.bmp")
+    image = pipe.generate(prompt,
+                          adapters=openvino_genai.AdapterConfig(),
+                          # passing adapters in generate overrides adapters set in the constructor; openvino_genai.AdapterConfig() means no adapters
```
[the remark this answers is not in the document]
Comment (popovaan) [reply samples/python/text2image/lora.py:60]: Done.

File: samples/python/text2image/main.py
[the remark this answers is not in the document]
Comment (popovaan) [reply samples/python/text2image/main.py:1]: Done.

File: samples/python/text2image/text2image.py [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+        random_generator=random_generator,
+    )
```
[the remark this answers is not in the document]
Comment (popovaan) [reply samples/python/text2image/text2image.py]: Done.

**Review by ilya-lavrenov — APPROVED** [review]
Good job!

File: .github/workflows/lcm_dreamshaper_cpp.yml [hunk 13]
```diff
@@ -42,7 +42,7 @@ jobs:
         run: |
           source ${{ env.OV_INSTALL_DIR }}/setupvars.sh
           cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
-          cmake --build ./build/ --config Release --parallel
+          cmake --build ./build/ --config Release --target stable_diffusion py_generate_pipeline --parallel
```
Comment (ilya-lavrenov) [inline .github/workflows/lcm_dreamshaper_cpp.yml:45]:
```suggestion
          cmake --build ./build/ --config Release --target stable_diffusion lora_stable_diffusion py_generate_pipeline --parallel
```

File: .github/workflows/lcm_dreamshaper_cpp.yml [hunk 14]
```diff
@@ -92,7 +100,7 @@ jobs:
         run: |
           . "${{ env.OV_INSTALL_DIR }}/setupvars.ps1"
           cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
-          cmake --build ./build/ --config Release --parallel
+          cmake --build ./build/ --config Release --target stable_diffusion py_generate_pipeline --parallel
```
Comment (ilya-lavrenov) [inline .github/workflows/lcm_dreamshaper_cpp.yml:103]:
```suggestion
          cmake --build ./build/ --config Release --target stable_diffusion lora_stable_diffusion py_generate_pipeline --parallel
```

File: .github/workflows/stable_diffusion_1_5_cpp.yml [hunk 15]
```diff
@@ -42,7 +42,7 @@ jobs:
         run: |
           source ${{ env.OV_INSTALL_DIR }}/setupvars.sh
           cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
-          cmake --build ./build/ --config Release --parallel
+          cmake --build ./build/ --config Release --target stable_diffusion py_generate_pipeline --parallel
```
Comment (ilya-lavrenov) [inline .github/workflows/stable_diffusion_1_5_cpp.yml:45]:
```suggestion
          cmake --build ./build/ --config Release --target stable_diffusion lora_stable_diffusion py_generate_pipeline --parallel
```

File: .github/workflows/stable_diffusion_1_5_cpp.yml [hunk 16]
```diff
@@ -98,7 +113,7 @@ jobs:
         run: |
           . "${{ env.OV_INSTALL_DIR }}/setupvars.ps1"
           cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
-          cmake --build ./build/ --config Release --parallel
+          cmake --build ./build/ --config Release --target stable_diffusion py_generate_pipeline --parallel
```
Comment (ilya-lavrenov) [inline .github/workflows/stable_diffusion_1_5_cpp.yml:116]:
```suggestion
          cmake --build ./build/ --config Release --target stable_diffusion lora_stable_diffusion py_generate_pipeline --parallel
```

File: .github/workflows/lcm_dreamshaper_cpp.yml [hunk 17]
```diff
@@ -69,6 +69,14 @@ jobs:
           source ${{ env.OV_INSTALL_DIR }}/setupvars.sh
           ./build/samples/cpp/text2image/stable_diffusion ./models/lcm_dreamshaper_v7/FP16 "cyberpunk cityscape like Tokyo New York  with tall buildings at dusk golden hour cinematic lighting"
 
+      - name: Run Python app
+        run: |
+          source openvino_lcm_cpp/bin/activate
+          source ./ov/setupvars.sh
+          export PYTHONPATH=./build/:$PYTHONPATH
```
Comment (ilya-lavrenov) [inline .github/workflows/lcm_dreamshaper_cpp.yml:76]:
```suggestion
          export PYTHONPATH=./build/:$PYTHONPATH
          export LD_LIBRARY_PATH=./build/openvino_genai:$LD_LIBRARY_PATH
```

File: .github/workflows/stable_diffusion_1_5_cpp.yml [hunk 18]
```diff
@@ -74,6 +74,21 @@ jobs:
         run: |
           source ${{ env.OV_INSTALL_DIR }}/setupvars.sh
           ./build/samples/cpp/text2image/lora_stable_diffusion ./models/dreamlike-art-dreamlike-anime-1.0/FP16 "curly-haired unicorn in the forest, anime, line" ./models/soulcard.safetensors 0.7
+      
+      - name: Run Python main app
+        run: |
+          source openvino_sd_cpp/bin/activate
+          source ./ov/setupvars.sh
+          export PYTHONPATH=./build/:$PYTHONPATH
```
Comment (ilya-lavrenov) [inline .github/workflows/stable_diffusion_1_5_cpp.yml:82]:
```suggestion
          export PYTHONPATH=./build/:$PYTHONPATH
          export LD_LIBRARY_PATH=./build/openvino_genai:$LD_LIBRARY_PATH
```

File: .github/workflows/stable_diffusion_1_5_cpp.yml [hunk 19]
```diff
@@ -74,6 +74,21 @@ jobs:
...
+          python ./samples/python/text2image/main.py ./dreamlike_anime_1_0_ov/FP16 "cyberpunk cityscape like Tokyo New York with tall buildings at dusk golden hour cinematic lighting"
+          
+      
+      - name: Run Python LoRA app
+        run: |
+          source openvino_sd_cpp/bin/activate
+          source ./ov/setupvars.sh
+          export PYTHONPATH=./build/:$PYTHONPATH
```
Comment (ilya-lavrenov) [inline .github/workflows/stable_diffusion_1_5_cpp.yml:90]:
```suggestion
          export PYTHONPATH=./build/:$PYTHONPATH
          export LD_LIBRARY_PATH=./build/openvino_genai:$LD_LIBRARY_PATH
```

File: src/python/py_text2image_pipeline.cpp [hunk 20]
```diff
@@ -0,0 +1,258 @@
...
+        py::arg("model_path"), "folder with exported model files.", 
+        R"(
+            Text2ImagePipeline class constructor.
+            model_path (str): Path to the folder with exported model files.
+        )")
+
+        .def(py::init([](
+            const std::string& model_path, 
+            const std::string& device,
+            const py::kwargs& kwargs
```
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/python/py_text2image_pipeline.cpp:182]: Applied in PR https://github.com/openvinotoolkit/openvino.genai/pull/1042

**Pull request merged**